src/Data/Bool/Properties.agda

```diff
@@ -586,6 +586,10 @@ not-involutive : Involutive not
 not-involutive true  = refl
 not-involutive false = refl
 
+not-injective : ∀ {x y} → not x ≡ not y → x ≡ y
+not-injective {false} {false} nx≢ny = refl
```

Comment:
This proof is strict on the implicit arguments but lazy on the explicit argument - is this really what we want?

Reply:
AGDA rejects being strict in the explicit argument without being strict in implicit here. Do you want me to make it strict in all arguments?

Reply:
Can you not match just on 'refl' ?

Reply:
I don't think so:

```
not-injective refl = refl
```

Results in:

```
I'm not sure if there should be a case for the constructor refl,
because I get stuck when trying to solve the following unification
problems (inferred index ≟ expected index):
  not x ≟ not y
when checking that the pattern refl has type not x ≡ not y
```

Reply:
You mean:

```agda
not-injective {false} {false} refl = refl
not-injective {true}  {true}  refl = refl
```

That's what I meant by being strict in all arguments.

Reply:
I was just incorrect here.  not-injective really does need to pattern-match on its boolean inputs.  I think your original definition is fine, sorry for the noise.